pandas/core/arrays/base.py

```diff
@@ -216,6 +216,18 @@ def isna(self):
         """
         raise AbstractMethodError(self)
 
+    def unique(self):
+        """Compute the ExtensionArray of unique values.
+
+        Returns
+        -------
+        uniques : ExtensionArray
+        """
+        from pandas import unique
+
+        uniques = unique(self.astype(object))
+        return type(self)(uniques)
```

Comment:
i think you should add a 
self._shallow_copy or something instead of doing this)

Reply:
To avoid the copy or to avoid the `type` stuff?

In this specific case, I think a copy is necessary since `uniques` is an ndarray objects.

As for alternative constructors to avoid the `type` stuff, sure, just need to come up with names for them.

For my IPAddress stuff I want to be able to do zero-copy construction from

- ExtensionArray
- numpy structured array with the correct fields.

Putting those in separate constructors makes sense.

Reply:
no its the idiom , see below, should have a ``_constructor`` in EA which just returns ``type(self)``, which is what we do in Index. this handled the sub-classing things.

Reply:
@jreback What do you mean with "this handled the sub-classing things"

Reply:
because *everywhere* else where use ``_constructor`` as the constructor class and do not use ``type(self)`` directly (yes that is the implementation, but the code all uses ``_constructor``

Reply:
I understand that, but it's not because we do it everywhere else *internally* that we need to do it here as well. Here we have something that is part of an external interface, and if adding something like that we should have a reason for it. And a clear message to the implementer what we expect from it. 
Eg in Series/DataFrame you have interactions between different subclasses (a slice of DataFrame can give another Series subclass). That's not something we have to deal with here.

Reply:
This just becomes confusing. I don't any reason not to have a ``_constructor``, its obvious, consistent in the code. These ad-hoc breaks with internal consistency are just creating technical debt. Please let's have consistentcy.

Reply:
You want me to change it to `_constructor` even though it'll be irrelevant after https://github.com/pandas-dev/pandas/pull/19913 is resolved?

Reply:
see my comments on that PR

Reply:
so this now should change to ``_construct_from_sequence``, in fact should be try to be concistent and *always* use this.